staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go

```diff
@@ -266,15 +274,21 @@ func New(opts Options) (*Authenticator, error) {
 	}
 
 	verifierConfig := &oidc.Config{
-		ClientID:             opts.JWTAuthenticator.Issuer.Audiences[0],
```

Comment:
it looks much cheaper to let go-oidc check the audience, since if it is incorrect, it returns an error before checking signatures

should we only do custom audience validation in the case where we have multiple audiences in our config (which I expect to be a much less common configuration)?

Reply:
> should we only do custom audience validation in the case where we have multiple audiences in our config (which I expect to be a much less common configuration)?

I don't have a strong opinion on this. By setting `SkipIDClientIDCheck` to `true`, I was thinking we consistently validate the audiences in the authenticator in 1 place.

@enj WDYT?

Reply:
My expectation was that we would only set `SkipIDClientIDCheck` to `true` when we had multiple audiences.  I wasn't concerned about performance, and expected us to perform our new manual audience check _always_ as well.  The rationale being 'we have to screw up two separate checks before we get a CVE.'

Reply:
sounds good! I've updated the behavior to match @enj's comment.